acvm-repo/acvm/src/pwg/memory_op.rs

```diff
@@ -74,11 +83,29 @@ impl<F: AcirField> MemoryOpSolver<F> {
         Ok(())
     }
 
+    /// Update the 'block_values' by processing the provided Memory opcode
+    /// The opcode 'op' contains the index and value of the operation and the type
+    /// of the operation.
+    /// They are all stored as 'Expressions'
+    /// The type of 'operation' is '0' for a read and '1' for a write. It must be a constant
+    /// expression.
+    /// Index is not required to be constant but it must reduce to a known value
+    /// for processing the opcode. This is done by doing the (partial) evaluation of its expression,
+    /// using the provided witness map.
+    ///
+    /// READ: read the block at index op.index and update op.value with the read value
+    /// - 'op.value' must reduce to a witness (after the evaluation of its expression)
+    /// - the value is updated in the provided witness map, for the 'op.value' witness
+    ///
+    /// WRITE: update the block at index 'op.index' with 'op.value'
+    /// - 'op.value' must reduce to a known value
+    ///
+    /// If a requirement is not met, it returns an error.
     pub(crate) fn solve_memory_op(
         &mut self,
         op: &MemOp<F>,
         initial_witness: &mut WitnessMap<F>,
-        predicate: &Option<Expression<F>>,
+        predicate: &Option<Expression<F>>, //TODO il faut enlver le predicat !?
```

Comment:
```suggestion
        predicate: &Option<Expression<F>>,
```
leftover?

Reply:
Thanks! I will work on removing this predicate right now.